File: c-fizzbuzz/fizz-buzz.test.c
#include <stdio.h>
#include <string.h>

typedef const struct spec {
  const int divisor;
  const char *const code;
} spec_t;

spec_t specs[] = {
    {3, "Fizz"},
    {5, "Buzz"},
    {7, "Whizz"},
    {11, "Bang"},
};

const int num_of_specs = sizeof(specs) / sizeof(specs[0]);

const int is_multiple(const int n, const int of) { return n % of == 0; }

void fizzbuzz(char *const output, const int n) {
  output[0] = '\0';

  for (spec_t *spec = specs; spec < specs + num_of_specs; spec++) {
    if (is_multiple(n, spec->divisor)) {
      // append the code
      strcat(output, spec->code);
    }
  }

  const int no_divisors = output[0] == '\0';

  if (no_divisors) {
    sprintf(output, "%d", n);
  }
}

int main(void) {
  const int least_common_multiple = 3 * 5 * 7 * 11;

  for (int n = 1; n <= least_common_multiple; n++) {
    char result[20];
    fizzbuzz(result, n);
    printf("%d -> %s\n", n, result);
  }
}

File: tcr-timer/fizzbuzz.test.ts
const fizz = { divisor: 3, code: "Fizz" };
const buzz = { divisor: 5, code: "Buzz" };

const specs = [fizz, buzz];

const fizzbuzz = (n: number) =>
    specs
        .filter(({ divisor }) => n % divisor == 0)
        .map(({ code }) => code)
        .join("") || String(n);
range(35)
    .map((x) => x + 1)
    .forEach((n) => console.log(fizzbuzz(n)));

function range(e: number) {
    return Array.from(Array(e)).map((_, i) => i);
}
